src/NetscriptFunctions/Singularity.ts

```diff
@@ -622,6 +622,31 @@ export function NetscriptSingularity(
         return Promise.resolve();
       });
     },
+    isFocused: function(): boolean {
+      helper.updateDynamicRam("isFocused", getRamCost("isFocused"));
+      helper.checkSingularityAccess("isFocused", 2);
+      return player.focus;
+    },
+    setFocus: function(focus: boolean): boolean {
+      helper.updateDynamicRam("setFocus", getRamCost("setFocus"));
+      helper.checkSingularityAccess("setFocus", 2);
+      if (!player.isWorking) {
+        throw helper.makeRuntimeErrorMsg("setFocus", "Not currently working");
+      }
+      if (!(player.workType == CONSTANTS.WorkTypeFaction || player.workType == CONSTANTS.WorkTypeCompany || player.workType == CONSTANTS.WorkTypeCompanyPartTime)) {
+        throw helper.makeRuntimeErrorMsg("setFocus", "Cannot change focus for current job");
+      }
+      if (!player.focus && focus) {
+        player.startFocusing();
+        Router.toWork();
+        return true;
+      } else if (player.focus && !focus) {
+        player.stopFocusing();
+        Router.toTerminal();
```

Comment:
I was questioning doing any routing at all on `setFocus(false)`, but I see now that we will *only* route to terminal if we were currently focused (e.g. on a work screen) so this makes sense.